File: mobile/ios/packages/town-crier-domain/Sources/ValueObjects/UserProfile.swift
/// A user's identity information from the authentication provider.
public struct UserProfile: Equatable, Sendable {
  public let userId: String
  public let email: String
  public let name: String?

  public init(userId: String, email: String, name: String? = nil) {
    self.userId = userId
    self.email = email
    self.name = name
  }

  /// Derives the authentication method from the userId prefix.
  public var authMethod: AuthMethod {
    if userId.hasPrefix("auth0|") {
      return .emailPassword
    } else if userId.hasPrefix("google-oauth2|") {
      return .google
    } else if userId.hasPrefix("apple|") {
      return .apple
    } else {
      return .unknown
    }
  }
}

File: mobile/android/domain/src/main/kotlin/uk/towncrierapp/domain/auth/UserProfile.kt
package uk.towncrierapp.domain.auth

/**
 * A user's identity information from the authentication provider. [email]
 * may legitimately be blank — Sign in with Apple grants no email scope — so
 * callers must render it gracefully rather than treat blank as an error.
 */
public data class UserProfile(
    val userId: String,
    val email: String,
    val name: String?,
) {
    /** Derives the authentication method from the [userId] (JWT `sub` claim) prefix. */
    public val authMethod: AuthMethod
        get() =
            when {
                userId.startsWith("auth0|") -> AuthMethod.EMAIL_PASSWORD
                userId.startsWith("google-oauth2|") -> AuthMethod.GOOGLE
                userId.startsWith("apple|") -> AuthMethod.APPLE
                else -> AuthMethod.UNKNOWN
            }
}
